airflow/models/serialized_dag.py

```diff
@@ -102,7 +102,7 @@ def __init__(self, dag: DAG, processor_subdir: str | None = None):
         dag_data = SerializedDAG.to_dict(dag)
         dag_data_json = json.dumps(dag_data, sort_keys=True).encode("utf-8")
 
-        self.dag_hash = hashlib.md5(dag_data_json).hexdigest()
+        self.dag_hash = hashlib.new("md5", data=dag_data_json, usedforsecurity=False).hexdigest()
```

Comment:
```suggestion
        self.dag_hash = hashlib.md5(dag_data_json, usedforsecurity=False).hexdigest()
```

The argument is available in all algorithms in hashlib, and according to the documentation:

> The named constructors are much faster than `new()` and should be preferred.

https://docs.python.org/3/library/hashlib.html#hashlib.new

Reply:
Also we cannot use it unconditionally. usedforsecurity is available as of airflow 3.9 and we are supporting 3.7+.
So if we want to add it, we need to add a compat library that will use it properly depending on Python version.

Reply:
Ah okay. I went with "new" since it used kwargs, so passing usedforsecurity would either work or just get ignored. But I see the precommit linting fails on that being unexpected.
Are there any examples currently of where it makes decisions based on the python version?

Reply:
Look for PY38 for example